content/cloud-docs/api-docs/providers.mdx

```diff
@@ -254,89 +318,1138 @@ curl \
+
+| Key path                      | Type   | Default | Description                                                                                  |
+| ----------------------------- | ------ | ------- | -------------------------------------------------------------------------------------------- |
+| `data.type`                   | string |         | Must be `"gpg-keys"`.                                                                        |
+| `data.attributes.namespace`   | string |         | The namespace of the provider. Must be the same as the `organization_name` for the provider. |
+| `data.attributes.ascii-armor` | string |         | A valid gpg-key string.                                                                      |
+
+### Sample Payload
+
+```json
+{
+  "data": {
+    "type": "gpg-keys",
+    "attributes": {
+      "namespace": "hashicorp",
+      "ascii-armor": "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\nmQINB...=txfz\n-----END PGP PUBLIC KEY BLOCK-----\n"
+    }  }
+}
+```
+
+### Sample Request
+
+```shell
+curl \
   --header "Authorization: Bearer $TOKEN" \
   --header "Content-Type: application/vnd.api+json" \
-  https://app.terraform.io/api/v2/organizations/my-organization/registry-providers/public/hashicorp/aws
+  --request POST \
+  --data @payload.json \
+  https://app.terraform.io/api/registry/private/v2/gpg-keys
 ```
 
-### Sample Response (public provider)
+### Sample Response
 
 ```json
 {
   "data": {
-    "id": "prov-fZn7uHu99ZCpAKZJ",
-    "type": "registry-providers",
+    "type": "gpg-keys",
+    "id": "23",
     "attributes": {
-      "name": "aws",
+      "ascii-armor": "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\nmQINB...=txfz\n-----END PGP PUBLIC KEY BLOCK-----\n",
+      "created-at": "2022-02-11T19:16:59Z",
+      "key-id": "32966F3FB5AC1129",
       "namespace": "hashicorp",
-      "registry-name": "public",
-      "created-at": "2020-07-09T19:36:56.288Z",
-      "updated-at": "2020-07-09T20:16:20.538Z",
-      "permissions": {
-        "can-delete": true
-      }
+      "source": "",
+      "source-url": null,
+      "trust-signature": "",
+      "updated-at": "2022-02-11T19:16:59Z"
     },
-    "relationships": {
-      "organization": {
-        "data": {
-          "id": "my-organization",
-          "type": "organizations"
-        }
-      }
+    "links": {
+      "self": "/v2/gpg-keys/23"
+    }
+  }
+}
+```
+
+## Get GPG Key (Private Provider)
+
+`GET /api/registry/:registry_name/v2/gpg-keys/:namespace/:key_id`
+
+### Parameters
+
+| Parameter            | Description                                          |
+| -------------------- | ---------------------------------------------------- |
+| `:registry_name`     | Must be `private`.                                   |
+| `:namespace`         | The namespace of the provider scoped to the GPG key. | 
+| `:key_id`            | The id of the GPG key.                               |
+
+Gets the content of a GPG key
```

Comment:
```suggestion
Gets the content of a GPG key.
```

Reply:
darnit, I really thought I got all of them lol